price-feeder/oracle/provider/okx.go

```
@@ -0,0 +1,212 @@
+	OkxTickerResponseWS struct {
+		Data []OkxTickerPair   `json:"data"`
+		Arg  SubscriptionTopic `json:"arg"`
+	}
+
+	SubscriptionTopic struct {
+		Channel string `json:"channel"` // Channel name ex.: tickers
+		InstId  string `json:"instId"`  // Instrument ID ex.: BTC-USDT
+	}
+
+	SubscriptionMsg struct {
+		Op   string              `json:"op"` // Operation ex.: subscribe
+		Args []SubscriptionTopic `json:"args"`
+	}
+)
+
+// NewOkxProvider creates a new OkxProvider
+func NewOkxProvider(ctx context.Context, pairs ...types.CurrencyPair) (*OkxProvider, error) {
+	wsURL := url.URL{
+		Scheme: "wss",
+		Host:   okxHost,
+		Path:   okxPath,
+	}
+
+	wsConn, _, err := websocket.DefaultDialer.Dial(wsURL.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to ws: %+v", err)
+	}
+
+	p := &OkxProvider{
+		wsURL:            wsURL,
+		wsClient:         wsConn,
+		tickersMap:       &sync.Map{},
+		msReadNewMessage: msReadNewMessage,
+	}
+
+	if err := p.newTickerSubscription(pairs...); err != nil {
+		return nil, err
+	}
+
+	go p.handleTickers(ctx)
+
+	return p, nil
+}
+
+// GetTickerPrices returns the tickerPrices based on the saved map
+func (p OkxProvider) GetTickerPrices(pairs ...types.CurrencyPair) (map[string]TickerPrice, error) {
+	tickerPrices := make(map[string]TickerPrice, len(pairs))
+
+	for _, currencyPair := range pairs {
+		cp := currencyPair
+		price, err := p.getTickerPrice(cp)
+		if err != nil {
+			return nil, err
+		}
+
+		tickerPrices[cp.String()] = price
+	}
+
+	return tickerPrices, nil
+}
+
+func (p OkxProvider) getTickerPrice(cp types.CurrencyPair) (TickerPrice, error) {
+	instrumentId := getInstrumentId(cp)
+	tickerPair, exist := p.getMapTicker(instrumentId)
+	if !exist { // it should exist the ticker will be
+		return TickerPrice{}, fmt.Errorf("ticker pair not found %s", instrumentId)
+	}
+
+	return tickerPair.ToTickerPrice()
+}
+
+func (p OkxProvider) getMapTicker(instrumentId string) (pair OkxTickerPair, exists bool) {
```

Should return an error instead, and probably use unnamed return vars since the other functions do : 
```suggestion
func (p OkxProvider) getMapTicker(instrumentId string) (OkxTickerPair, error) {
```